### PR TITLE
Fix wandb logger on resume

### PR DESCRIPTION
**Pull request by Louis-Dupont**

### CODE REVIEW

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 1]
```diff
@@ -1610,24 +1619,25 @@ def _get_hyper_param_config(self):
...
         if self.training_params.save_model:
             self._save_checkpoint(self.optimizer, epoch + 1, validation_results, context)
 
-    def _write_lrs(self, epoch):
+    def _get_lr_dict(self) -> dict:
         lrs = [self.optimizer.param_groups[i]["lr"] for i in range(len(self.optimizer.param_groups))]
         lr_titles = ["LR/Param_group_" + str(i) for i in range(len(self.optimizer.param_groups))] if len(self.optimizer.param_groups) > 1 else ["LR"]
         lr_dict = {lr_titles[i]: lrs[i] for i in range(len(lrs))}
-        self.sg_logger.add_scalars(tag_scalar_dict=lr_dict, global_step=epoch)
+        return deepcopy(lr_dict)
```
Comment (shaydeci) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:1640]: Why the deep copy?

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 2]
```diff
@@ -444,7 +446,7 @@ def _train_epoch(self, epoch: int, silent_mode: bool = False) -> tuple:
 
             # LOG LR THAT WILL BE USED IN CURRENT EPOCH AND AFTER FIRST WARMUP/LR_SCHEDULER UPDATE BEFORE WEIGHT UPDATE
```
Comment (shaydeci) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:447]: I think we need to update the comment here.

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 3]
```diff
@@ -194,6 +194,8 @@ def __init__(self, experiment_name: str, device: str = None, multi_gpu: Union[Mu
         self.max_train_batches = None
         self.max_valid_batches = None
 
+        self.epoch_start_lr_dict = {}
```
Comment (shaydeci) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:197]: seem to me that this is only for internal use, please change attribute name to _epoch_start_lr_dict.

**Review by shaydeci — CHANGES_REQUESTED** [review]

File: src/super_gradients/training/sg_trainer/sg_trainer.py [hunk 4]
```diff
@@ -444,7 +446,7 @@ def _train_epoch(self, epoch: int, silent_mode: bool = False) -> tuple:
 
             # LOG LR THAT WILL BE USED IN CURRENT EPOCH AND AFTER FIRST WARMUP/LR_SCHEDULER UPDATE BEFORE WEIGHT UPDATE
             if not self.ddp_silent_mode and batch_idx == 0:
-                self._write_lrs(epoch)
+                self.epoch_start_lr_dict = self._get_lr_dict()
```
Comment (BloodAxe) [inline src/super_gradients/training/sg_trainer/sg_trainer.py:449]: What if one day we decide we want to log not only LR but also a WD? 
I'm suggesting to give a more common name to this container where one can put values to be logged by the end of epoch.

Comment (shaydeci) [reply src/super_gradients/training/sg_trainer/sg_trainer.py:449]: nice one @BloodAxe

Comment (Louis-Dupont) [reply src/super_gradients/training/sg_trainer/sg_trainer.py]: I renamed like that, what do you guys think?
https://github.com/Deci-AI/super-gradients/blob/b175cb56639df38a5bdc90a7d4baf5fb0bed8965/src/super_gradients/training/sg_trainer/sg_trainer.py#L448

**Review by BloodAxe — APPROVED** [review]
LGTM

**Review by shaydeci — APPROVED** [review]
LGTM
